### PR TITLE
Begin adding a layman's-terms description of Endpoints' functionality

### PR DESCRIPTION
**Pull request by kadamwhite**
The existing "What Is It?" section did not fully explain to me how Endpoints was intended to be used, so I've taken a stab at articulating in a little more detail the purpose of the library and its relationship to your existing tools. I find these sorts of overviews to be personally useful, and would love input on how to improve the description so that it can help expand Endpoints' potential audience.

### CODE REVIEW

**Comment by kadamwhite** [comment]
I'd love to expand or clarify the text here, I see it as a quick intro that would complement the longer-form tutorials and instructions you were discussing earlier this week

**Comment by tkellen** [comment]
:+1: I think this is great!

File: README.md [hunk 1]
```diff
@@ -1,24 +1,30 @@
...
 
 [![NPM](https://nodei.co/npm/endpoints.png)](https://nodei.co/npm/endpoints/)
 
-# What is it?
-A hypermedia API framework compliant with the [json api](http://jsonapi.org) specification, compatible with any http server that supports express or hapi style request handlers.
+Endpoints is a hypermedia API framework compliant with the [json api](http://jsonapi.org) specification, compatible with any http server that supports express or hapi style request handlers.
+
+## What is it?
+
+Endpoints gives you a powerful set of tools for defining and structuring an API. Endpoints conforms to the JSON API standard, a set of conventions for writing efficient JSON-based APIs, so that your data can be transferred as efficiently as possible. Endpoints lets you define your API's routes and data model in a way that integrates with your existing web framework (*e.g.* [express](http://expressjs.com/) or [hapi](http://hapijs.com/)) and ORM (*e.g.* [Bookshelf](http://bookshelfjs.org/)).
```
Comment (tkellen) [inline README.md]: Can you remove one instance of efficient in this paragraph? *writing ~~efficient~~ JSON-based APIs*

**Comment by ashleygwilliams** [comment]
this is a great start at working on the wording. do you want to keep this as a WIP, @kadamwhite? if not, i'll merge.

**Comment by ashleygwilliams** [comment]
also, if you're done, if you could squash into a single commit with bullet points that'd be great :relaxed:

**Pull request closed without merging**